webrender_traits/src/display_list.rs

```diff
@@ -85,27 +116,309 @@ impl BuiltDisplayList {
         &self.descriptor
     }
 
-    pub fn all_display_items(&self) -> &[DisplayItem] {
-        unsafe {
-            convert_blob_to_pod(&self.data)
+    pub fn times(&self) -> (u64, u64) {
+      (self.descriptor.builder_start_time, self.descriptor.builder_finish_time)
+    }
+
+    pub fn iter(&self) -> BuiltDisplayListIter {
+        BuiltDisplayListIter::new(self)
+    }
+
+    pub fn get<T: Deserialize>(&self, range: ItemRange<T>) -> AuxIter<T> {
+        AuxIter::new(&self.data[range.start .. range.start + range.length])
+    }
+}
+
+impl<'a> BuiltDisplayListIter<'a> {
+    pub fn new(list: &'a BuiltDisplayList) -> Self {
+        BuiltDisplayListIter {
+            list: list,
+            data: &list.data,
+            // Dummy data, will be overwritten by `next`
+            cur_item: DisplayItem {
+                item: SpecificDisplayItem::PopStackingContext,
+                rect: LayoutRect::zero(),
+                clip_and_scroll: ClipAndScrollInfo::simple(ClipId::new(0, PipelineId(0, 0))),
+            },
+            cur_stops: ItemRange::default(),
+            cur_glyphs: ItemRange::default(),
+            cur_filters: ItemRange::default(),
+            cur_clip: ClipRegion::empty(),
+            peeking: Peek::NotPeeking,
         }
     }
 
-    pub fn into_display_items(self) -> Vec<DisplayItem> {
-        unsafe {
-            convert_vec_blob_to_pod(self.data)
+    pub fn display_list(&self) -> &'a BuiltDisplayList {
+        self.list
+    }
+
+    pub fn next<'b>(&'b mut self) -> Option<DisplayItemRef<'a, 'b>> {
+        use SpecificDisplayItem::*;
+
+        match self.peeking {
+            Peek::IsPeeking => {
+                self.peeking = Peek::NotPeeking;
+                return Some(self.as_ref())
+            }
+            Peek::StartPeeking => {
+                self.peeking = Peek::IsPeeking;
+            }
+            Peek::NotPeeking => { /* do nothing */ }
         }
+
+        // Don't let these bleed into another item
+        self.cur_stops = ItemRange::default();
+        self.cur_clip = ClipRegion::empty();
+
+        loop {
+            if self.data.len() == 0 { return None }
```

Comment:
We should probably use newlines here, the single line style is not something that we typically use in WebRender as far as I know.